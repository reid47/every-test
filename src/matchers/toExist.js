import { formatValue } from './common';

export default function toExist(received) {
  const isTruthy = !!received;
  const isHtmlElement = received instanceof HTMLElement;
  const isInDocument = document.contains(received);

  const pass = isTruthy && isHtmlElement && isInDocument;

  let message;
  if (pass) {
    message = 'Expected element not to exist in the DOM, but it did.';
  } else if (!isTruthy) {
    message = `Expected element to exist, but it did not.`;
  } else if (!isHtmlElement) {
    message = `Expected assertion to be made on a DOM element, but received ${formatValue(
      received
    )}.`;
  } else {
    message =
      'DOM element is not contained by document. Was this element rendered to the document?';
  }

  return { pass, message };
}
