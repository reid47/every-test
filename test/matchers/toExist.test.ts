import toExist from '../../src/matchers/toExist';

describe('toExist', () => {
  test('passing', () => {
    const el = document.createElement('div');
    document.body.appendChild(el);

    const result = toExist(el);
    expect(result.pass).toBeTruthy();
    expect(result.message).toMatchSnapshot();
  });

  test('failing (falsy)', () => {
    const el = null;

    const result = toExist(el);
    expect(result.pass).toBeFalsy();
    expect(result.message).toMatchSnapshot();
  });

  test('failing (not an HTML element)', () => {
    const el = 47;

    const result = toExist(el);
    expect(result.pass).toBeFalsy();
    expect(result.message).toMatchSnapshot();
  });

  test('failing (not in document)', () => {
    const el = document.createElement('div');

    const result = toExist(el);
    expect(result.pass).toBeFalsy();
    expect(result.message).toMatchSnapshot();
  });
});
