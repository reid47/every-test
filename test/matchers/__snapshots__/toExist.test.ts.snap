// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toExist failing (falsy) 1`] = `"Expected element to exist, but it did not."`;

exports[`toExist failing (not an HTML element) 1`] = `"Expected assertion to be made on a DOM element, but received 47."`;

exports[`toExist failing (not in document) 1`] = `"DOM element is not contained by document. Was this element rendered to the document?"`;

exports[`toExist passing 1`] = `"Expected element not to exist in the DOM, but it did."`;
